Gems/AtomLyIntegration/CommonFeatures/Code/Source/CoreLights/CapsuleLightDelegate.cpp

```diff
@@ -98,6 +99,22 @@ namespace AZ::Render
     Aabb CapsuleLightDelegate::GetLocalVisualizationBounds() const
     {
         const auto [radius, height] = CalculateCapsuleVisualizationDimensions();
-        return Aabb::CreateCenterRadius(Vector3::CreateZero(), GetMax(radius, height));
+        return Aabb::CreateFromMinMax(
+            AZ::Vector3(-radius, -radius, AZ::GetMin(-radius, -height * 0.5f)),
+            AZ::Vector3(radius, radius, AZ::GetMax(radius, height * 0.5f)));
+    }
```

Comment:
It's unfortunate that `ShapeComponentRequestBus` has a `GetTransformAndLocalBounds` for `CapsuleShape` which is very similar to this. Longer term we may want some common functions for generating aabbs of various core math shapes - then we'd just create an `AZ::Capsule` for the light bounds and call a function to turn it into an aabb. This is fine for now though, just wanted to plant a seed :)

Reply:
I agree utility functions for match operations like this would be great. I would ideally like us to remove `GetTransformAndLocalBounds` from the `ShapeRequestBus` and instead use `GetWorldBounds` and `GetLocalBounds` on the `BoundRequestBus` (different components can then implement this as needed which is what we're doing here). Having the util function the bus call can delegate too would definitely be nice though. Hopefully we can get to it at some point 🙂